### PR TITLE
Upgrade `mkdocs-material`

### PR DESCRIPTION
**Pull request by jtherrmann**
See https://github.com/ASFHyP3/mkdocs-asf-theme/issues/27 for context

How to test these changes locally:

* `cd` into your local clone of `mkdocs-asf-theme`.
* Pull the latest changes and switch to the `upgrade-mkdocs-material` branch.
* `cd` into your local clone of `hyp3-docs`.
* Pull the latest changes and switch to the [`upgrade-theme`](https://github.com/ASFHyP3/hyp3-docs/tree/upgrade-theme) branch.
* Activate the `hyp3-docs` virtual environment and run `pip install -e ../mkdocs-asf-theme/`.
* Run `mkdocs serve`.

You can make changes to `mkdocs-asf-theme` and then display them on the `hyp3-docs` website by `Ctrl+C`ing and re-running `mkdocs serve`.

### CODE REVIEW

**Comment by jtherrmann** [comment]
I confirmed that all elements and classes in `asf.css` are up-to-date with the new HTML files.

**Comment by jtherrmann** [comment]
Moving remaining TODOs to https://github.com/ASFHyP3/mkdocs-asf-theme/issues/39

**Comment by jtherrmann** [comment]
Moving remaining style differences to https://github.com/ASFHyP3/mkdocs-asf-theme/issues/40

File: setup.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
     install_requires=[
         'mkdocs',
-        'mkdocs-material>=6.2.1,<7.0',
+        'mkdocs-material==9.4.2',
```
Comment (jhkennedy) [inline setup.py:34]: @jtherrmann do we want to be this strict with the material version, or constrain to the major version?

**Review by jhkennedy — APPROVED** [review]
Looks good for develop -- I think we'll want to have a discussion about the range of mkdocs-material we support before releasing.

File: setup.py (hunk 1 above)
Comment (jtherrmann) [reply setup.py:34]: No idea, we should discuss this further.

**Pull request merged**